### ISSUE TITLE
Questions about Performance Optimization

### ISSUE THREAD

**Issue by maxin9966** [camenduru/flux-jupyter#3]
I have two questions, could you answer them please:

1. In dev, the model used is int8, and my tests are excellent. However, in schnall, the model is not int8, does this mean it will consume more GPU memory?
2. In the dev script, when tested on a 16G GPU, it can generate a 1536*1536 image. I notice that each time an image is generated, it takes 10 seconds to reload the model. Is it possible to keep the model resident to reduce the time it takes to generate each image? Since my workflow only requires a 1024*1024 image, I would like to optimize for speed by trading space for time, aiming to increase the speed of image generation. How can I optimize the code?

**Comment by maxin9966** [comment]
def generate(prompt, steps, guidance, width, height, seed):
    with torch.inference_mode():
        if width > 1536:
            width = 1536
        if height > 1536:
            height = 1536

        max_pixels = 1200 * 1200
        if width * height > max_pixels:
            ratio = math.sqrt(max_pixels / (width * height))
            width = int(width * ratio)
            height = int(height * ratio)
            print(f"{width}x{height}")

        width = int((16 * width) // 16)
        height = int((16 * height) // 16)
        # seed = -1
        # steps = 20
        if steps > 20:
            steps = 20
        sampler_name = "euler"
        scheduler = "simple"

        if seed == -1:
            seed = random.randint(0, 18446744073709551615)
        print(seed)

        positive_prompt = translate_text_if_not_english(prompt)
        cond, pooled = clip.encode_from_tokens(clip.tokenize(positive_prompt), return_pooled=True)
        cond = [[cond, {"pooled_output": pooled}]]
        noise = RandomNoise.get_noise(seed)[0]
        guider = BasicGuider.get_guider(unet, cond)[0]
        sampler = KSamplerSelect.get_sampler(sampler_name)[0]
        sigmas = BasicScheduler.get_sigmas(unet, scheduler, steps, 1.0)[0]
        latent_image = EmptyLatentImage.generate(closestNumber(width, 16), closestNumber(height, 16))[0]
        sample, sample_denoised = SamplerCustomAdvanced.sample(noise, guider, sampler, sigmas, latent_image)
        model_management.soft_empty_cache()
        decoded = VAEDecode.decode(vae, sample)[0].detach()
        # Image.fromarray(np.array(decoded*255, dtype=np.uint8)[0]).save("./output/flux.png")
        image = Image.fromarray(np.array(decoded * 255, dtype=np.uint8)[0])
        image_path = compress_image(image)
        return image_path

import gradio as gr

demo = gr.Interface(fn=generate, inputs=["textbox", gr.Slider(8, 20, value=20, step=1), gr.Number(value=3.5), gr.Slider(384, 1536, value=1024, step=2), gr.Slider(384, 1536, value=1024, step=2), gr.Number(value=-1)], outputs="image")
demo.queue().launch(server_name='0.0.0.0', server_port=7865, show_api=False)
